server/routes/api/auth.js

```diff
@@ -9,26 +13,55 @@ const User = require('../../models/user');
 router.post('/login', (req, res) => {
   User.authenticate(req.body.email, req.body.password)
     .then(
-      user => {
+      (user) => {
         req.session.userId = user._id;
         return res.status(200).json(user);
       },
-      err => {
-        return res.status(400).json(err);
-      }
+      err => res.status(400).json(err),
     )
-    .catch(err => {
-      return res.status(500).json(err);
-    });
+    .catch(err => res.status(500).json(err));
 });
 
 router.get('/logout', (req, res) => {
-  if (req.session.userId) {
-    req.session.destroy(err => {
-      err ? res.status(500).json(err) : res.status(200)
-    })
+  req.session.destroy(err => (err ? res.status(500).json(err) : res.status(200).end()));
+});
+
+router.post('/forgot', async (req, res) => {
+  const user = await User.findOne({ email: req.body.email }).exec();
+  if (!user) {
+    return res.status(200).end();
   }
-  res.end()
-})
+  const token = crypto.randomBytes(20).toString('hex');
+  user.resetPassword.token = token;
+  user.resetPassword.expiration = Date.now() + 3600000;
+  try {
+    await user.save();
+  } catch (err) {
+    return res.status(500).json({ error: err });
+  }
+
+  const smtpTransport = nodemailer.createTransport({
+    host: 'smtp.sendgrid.net',
+    port: 587,
+    auth: {
+      user: 'apikey',
+      pass: process.env.SG_PASS,
+    },
+  });
+
+  const mailOptions = {
+    from: process.env.PASSWORD_RESET_AUTH_EMAIL,
+    to: user.email,
+    subject: 'Forgot Password Request',
+    html: `<p>You are receiving this because you, or someone else, requested a password reset. Click <a href="http://localhost:${process.env.PORT}/reset/${token}">here</a> to finish resetting your password.</p>`,
+  };
+  try {
+    await smtpTransport.sendMail(mailOptions);
+    return res.status(200).end();
+  } catch (err) {
+    return res.status(500).json({ error: err });
```

Comment:
FYI, changing the sendgrid pw to something else gives us an authentication error.
```
{
    "error": {
        "code": "EAUTH",
        "response": "535 Authentication failed: The provided authorization grant is invalid, expired, or revoked",
        "responseCode": 535,
        "command": "AUTH PLAIN"
    }
}
```
I actually think sending back a 500 is appropriate here. I'm just concerned _what_ errors we may receive from sendgrid. I'm wondering if it's safer (and less misleading to the user) to not send back the error. Let's _just_ send back the status code. Eventually, as we talked about logging these errors, we can just store this error for us to know about.

Reply:
Okay. I removed the error message. It just send back a status of 500 now.